### PR TITLE
[IMP] website_forum: refactor the forum post route

### PR DESCRIPTION
**Pull request by yava-odoo**
In this PR, the get_post_values method is created and later overridden in the enterprise version

task-3475455

### CODE REVIEW

**Review by xavierbol — COMMENTED** [review]
Why not overriding the `_prepare_user_values` in our module instead of creating a new hook to override it in our module? 🤔

**Comment by yava-odoo** [comment]
Hello @xavierbol 
We can override the function but we want show_create_ticket in only one template
If we override _prepare_user_values then we send show_create_ticket in every template
Which is unnecessary I think and I'll little bit increase the computation also :thinking: 
That's why I've created another function to override in the ENT

**Review by xavierbol — COMMENTED** [review]
Thanks again for your work. 🙂 

I left 2 last comments and after that it will look good to me. 🙂

File: addons/website_forum/controllers/website_forum.py [hunk 1]
```diff
@@ -311,17 +321,9 @@ def question(self, forum, question, **post):
...
-            'main_object': question,
-            'edit_in_backend': True,
-            'question': question,
-            'header': {'question_data': True},
-            'filters': filters,
-            'reversed': reversed,
-        })
-
+        post_values = self.get_post_values(question, values)
+        values.update(post_values)
```
[the remark this answers is not in the document]
Comment (yava-odoo) [reply addons/website_forum/controllers/website_forum.py]: Hello @xavierbol 
Done :+1:

**Comment by xavierbol** [comment]
It seems you have a conflict, can you rebase? 

Thanks in advance!

**Review by xavierbol — COMMENTED** [review]
LGTM, thanks for your work, I will let Website team do the final review. 🙂

**Review by xavierbol — APPROVED** [review]

**Comment by xavierbol** [comment]
Oh I thought it was the scope of the Website but in fact, it is more the one of the SM team. 🙂

**Review by tde-banana-odoo — APPROVED** [review]

**Comment by xavierbol** [comment]
robodoo r+

**Comment by xavierbol** [comment]
robodoo r+

File: addons/website_forum/controllers/website_forum.py [hunk 2]
```diff
@@ -311,19 +324,7 @@ def question(self, forum, question, **post):
         if question.parent_id:
             redirect_url = "/forum/%s/%s" % (slug(forum), slug(question.parent_id))
             return request.redirect(redirect_url, 301)
-        filters = 'question'
-        values = self._prepare_user_values(forum=forum, searches=post)
-        values.update({
-            'main_object': question,
-            'edit_in_backend': True,
-            'question': question,
-            'seo_microdata': question._get_microdata(),
```
Comment (xavierbol) [inline addons/website_forum/controllers/website_forum.py:320]: I don't think it is expected to remove that line, is it? wrong rebase?

File: addons/website_forum/tests/test_performance.py [hunk 3]
```diff
@@ -34,7 +34,7 @@ def test_perf_sql_forum_standard_data(self):
         number_of_queries = self._get_url_hot_query(self.forum._compute_website_url(), cache=False)
         self.assertLessEqual(number_of_queries, 28)
         number_of_queries = self._get_url_hot_query(self.post.website_url)
-        self.assertEqual(number_of_queries, 21)
+        self.assertEqual(number_of_queries, 20)
```
Comment (xavierbol) [inline addons/website_forum/tests/test_performance.py:37]: Now, you added again `'seo_microdata': question._get_microdata(),` line, I think we no longer need to change this line

Comment (yava-odoo) [reply addons/website_forum/tests/test_performance.py]: Hello @xavierbol :wave:

Apologies :pray: 
I've updated the code

**Review by xavierbol — COMMENTED** [review]
robodoo r+

**Pull request closed without merging**